app/main.py

```diff
@@ -1,3 +1,44 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: ("Distance", int, float)) -> callable:
+        if not isinstance(other, (int, float)):
+            return Distance(self.km + other.km)
+        else:
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: (int, float)) -> callable:
+        if not isinstance(other, (int, float)):
+            self.km += other.km
+            return self
+        else:
+            self.km += other
+            return self
+
+    def __mul__(self, other: int) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: ("Distance", int, float)) -> callable:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: ("Distance", int, float)) -> callable:
+        return self.km < other
+
+    def __gt__(self, other: ("Distance", int, float)) -> callable:
```

Comment:
The type hint for the 'other' parameter should be 'Union[Distance, int, float]' instead of a tuple. This will correctly indicate that 'other' can be of any of these types.